### ISSUE TITLE
Non-existent functions should not 'work' inside or clauses

### ISSUE THREAD

**Issue by refset** [xtdb/crux#2492]
Reported via https://juxt-oss.zulipchat.com/#narrow/stream/194466-xtdb-users/topic/non-existent.20functions.20work.20inside.20or.20clauses

```
;; It seems that function calls inside an or are let off the hook from normal existence checks?

;; sensibly throws `xtdb.IllegalArgumentException`:
;; Clause refers to unknown variable: clojure.core/foo?
;; {:pred {:pred-fn clojure.core/foo?, :args [a]}}
(xt/q (xt/db xtdb-node)
      '{:find [a]
        :where [[a :a]
                [(clojure.core/foo? a)]]})

;; surprisingly returns empty set without throwing
(xt/q (xt/db xtdb-node)
      '{:find [a]
        :where [[a :a]
                (or [(clojure.core/foo? a)]
                    [(clojure.core/bar? a)])]})
```

**Comment by refset** [comment]
Possibly related or dupe of https://github.com/xtdb/xtdb/issues/1937

**Comment by wotbrew** [comment]
Think this is 'fixed' with #2560 in that you will no longer see empty set, however I think I'll keep this open to improve the error message.

**Comment by wotbrew** [comment]
Leaving error messages as is for now. I had thought that the caching still caused a different error message to return, but I was wrong. Resolved by #2560
